### ISSUE TITLE
Example on using the extracted features for classification task

### ISSUE THREAD

**Issue by anushiya-thevapalan** [DeepSpectrum/DeepSpectrum#28]
Hi,

I read your paper and was interested in getting to know how we can feed the extracted features to a classifier. Do you have any implementation to directly feed it? Is your complete pipeline (DeepSpectrum feature extraction + SVM classifier) available opensource?

If not, do we have to concatenate all the vectors corresponding to an audio clip and feed it to the classifier (After fixing the input shape of the vector by padding or removing the excess?)

Do you suggest any better approach to use the extracted features for the classification problems?

Thanks in advance

**Comment by mauricege** [comment]
Hi,

it depends on your Audiofiles: If you have clips that are around 10 seconds or shorter, you should be able to get some decent results by just extracting one vector for each clip. If do it this way, you can try different classifiers (e.g. those included in sklearn or WEKA), directly on the extracted feature files. Each row of the file includes the name of the audio sample, the elements of the feature vector and a label (if you supplied this information). In our experiments, we have found that linear Support Vector Machines work quite well with the DeepSpectrum features, we mostly use them through scikit-learn.

If you have longer files, you can extract features from overlapping windows (with -t WINDOW_SECONDS HOP_SECONDS). You will then get feature files that have an additional timestamp column, specifying where the window of the current feature vector starts. To classify those kinds of features, you can either use some sort of statistical aggregation (taking the mean of the vectors or quantising them) or use machine learning models designed for time-series data, e.g. RNNs. For time-series models, you have to stack the vectors of each audiofile (and then maybe perform batch padding). Currently, you have to do these steps by hand as DeepSpectrum only provides feature extraction to csv and arff files.

Best wishes,
Maurice
